### PR TITLE
Add Combobox input value APIs

### PR DESCRIPTION
**Pull request by diegohaz**
## Motivation

Combobox currently uses `value` for the text in the input and `selectedValue` for the committed selection. With Combobox Select unifying the selection APIs, naming the text state `inputValue` creates a migration path that can eventually free `value` for selection without changing behavior today.

## Solution

Add `inputValue`, `defaultInputValue`, `setInputValue`, `resetInputValue`, and `ComboboxInputValue`. Keep the existing APIs as synchronized, JSDoc-deprecated aliases, with the new names taking precedence when both are provided.

```diff
-<ComboboxProvider value={query} setValue={setQuery}>
+<ComboboxProvider inputValue={query} setInputValue={setQuery}>
   <Combobox />
-  <ComboboxValue />
+  <ComboboxInputValue />
 </ComboboxProvider>
```

Update internal consumers and representative examples to the canonical names, while preserving legacy controlled callbacks, connected stores, default-state conflict validation, reset behavior, and keyboard focus.

## Review gate reassessment

<details>
<summary>Cycle 3: Continue the synchronized compatibility alias design</summary>

**Assessment**

The third finding was low severity but deterministic: every generated `ComboboxValue` reference lost its `store` and `children` props. This metadata is part of the supported soft-deprecation path, and its likely impact is on users consulting the legacy reference while migrating. A local typed wrapper and focused assertions restore it with minimal implementation and maintenance cost.

The synchronized state keys, controlled-prop precedence, connected-store handling, and conflict validation carry intrinsic complexity, but that complexity follows directly from preserving `value` as a supported alias. The first review cycle identified compatibility gaps in that contract; the next two cycles found static changeset and reference-metadata omissions rather than architectural instability.

**Decision**

Continue the current design and preserve `ComboboxValue` as a typed wrapper around `ComboboxInputValue`. Do not broaden the reference extractor, narrow the compatibility contract, or leave any known migration edge case unsupported.

</details>

## Verification

- `pnpm lint`
- `pnpm tsc`
- Core Combobox store tests
- Combobox React sandbox tests, including legacy tabs
- JSDoc and website reference metadata tests
- `pnpm changeset status`
- `pnpm build`
- `pnpm build-website-lite`
- `pnpm build-app-lite`
- Package dry-runs and public export resolution

### CODE REVIEW

File: .changeset/290-combobox-value.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@ariakit/components": patch
+"@ariakit/react-components": patch
+"@ariakit/react": patch
+---
+
+Deprecated the Combobox input [`value`](https://ariakit.com/reference/combobox-provider#value), [`defaultValue`](https://ariakit.com/reference/combobox-provider#defaultvalue), [`setValue`](https://ariakit.com/reference/combobox-provider#setvalue), and [`resetValue`](https://ariakit.com/reference/use-combobox-store#resetvalue) APIs and [`ComboboxValue`](https://ariakit.com/reference/combobox-value) in favor of [`inputValue`](https://ariakit.com/reference/combobox-provider#inputvalue), [`defaultInputValue`](https://ariakit.com/reference/combobox-provider#defaultinputvalue), [`setInputValue`](https://ariakit.com/reference/combobox-provider#setinputvalue), [`resetInputValue`](https://ariakit.com/reference/use-combobox-store#resetinputvalue), and [`ComboboxInputValue`](https://ariakit.com/reference/combobox-input-value).
```
Comment (diegohaz) [inline .changeset/290-combobox-value.md]: This is a soft-deprecation, so there's no need for this changeset.

File: packages/ariakit-components/src/combobox/combobox-store.ts [hunk 2]
```diff
@@ -414,7 +434,23 @@ export function createComboboxStore({
     }),
   );
 
+  const setInputValue: ComboboxStore["setInputValue"] = (value) => {
+    combobox.setState("inputValue", value);
+    combobox.setState("value", combobox.getState().inputValue);
```
Comment (diegohaz) [inline packages/ariakit-components/src/combobox/combobox-store.ts:439]: Why not this?
```suggestion
    combobox.setState("value", value);
```

I wonder if this would be impacted by https://github.com/ariakit/ariakit/issues/5695 in the future.

File: packages/ariakit-components/src/combobox/combobox-store.ts [hunk 3]
```diff
@@ -414,7 +434,23 @@ export function createComboboxStore({
...
+  const resetInputValue = () => setInputValue(initialState.inputValue);
+
   const setState: ComboboxStore["setState"] = (key, value) => {
+    if (key === "inputValue" || key === "value") {
+      combobox.setState(key, value);
+      if (key === "inputValue") {
+        combobox.setState("value", combobox.getState().inputValue);
+      } else {
+        combobox.setState("inputValue", combobox.getState().value);
+      }
```
Comment (diegohaz) [inline packages/ariakit-components/src/combobox/combobox-store.ts:451]: Same here

File: packages/ariakit-react-components/src/combobox/combobox-value.tsx [hunk 4]
```diff
@@ -1,63 +1,18 @@
...
- * @example
- * ```jsx {3-5}
- * <ComboboxProvider>
- *   <Combobox />
- *   <ComboboxValue>
- *     {(value) => `Current value: ${value}`}
- *   </ComboboxValue>
- * </ComboboxProvider>
- * ```
+ * @deprecated Use
```
Comment (diegohaz) [inline packages/ariakit-react-components/src/combobox/combobox-value.tsx:16]: This is just a soft-deprecation. We can keep the JSDoc description, see tag and examples. The deprecated tag is just an addition.

File: packages/ariakit-react-components/src/combobox/combobox.tsx [hunk 5]
```diff
@@ -232,14 +232,14 @@ export const useCombobox = createHook<TagName, ComboboxOptions>(
     const open = useStoreState(store, "open");
     const contentElement = useStoreState(store, "contentElement");
 
-    // The current input value may differ from state.value when
+    // The current input value may differ from state.inputValue when
     // autoComplete is either "both" or "inline", in which case it will be
     // the active item value or a combination of the input value and the active
     // item value if it's the first item and it's been auto selected. This will
     // only affect the element's value, not the combobox state.
     const value = useMemo(() => {
```
Comment (diegohaz) [inline packages/ariakit-react-components/src/combobox/combobox.tsx]: Should we rename this internal variable to `inputValue` for consistency? We plan to rename `selectedValue` to `value` in the future, and I don't want to create any confusion here.

File: .changeset/290-combobox-value.md (hunk 1 above)
Comment (diegohaz) [reply .changeset/290-combobox-value.md]: Fixed in [`00cfb23`](https://github.com/ariakit/ariakit/commit/00cfb233767cb5d363a0f9a9c633800c172ae2e8): removed the deprecation-only changeset while keeping the feature changeset.

File: packages/ariakit-components/src/combobox/combobox-store.ts (hunk 2 above)
Comment (diegohaz) [reply packages/ariakit-components/src/combobox/combobox-store.ts:439]: Tracked in [#5695](https://github.com/ariakit/ariakit/issues/5695): forwarding a functional updater to both aliases in the current synchronous model double-applies it through the alias listeners, while a reentrant-safe paired write requires queued request semantics. That belongs to the planned request-versus-commit redesign rather than this soft-deprecation PR.

File: packages/ariakit-components/src/combobox/combobox-store.ts (hunk 3 above)
Comment (diegohaz) [reply packages/ariakit-components/src/combobox/combobox-store.ts:451]: Tracked in [#5695](https://github.com/ariakit/ariakit/issues/5695): forwarding a functional updater to both aliases in the current synchronous model double-applies it through the alias listeners, while a reentrant-safe paired write requires queued request semantics. That belongs to the planned request-versus-commit redesign rather than this soft-deprecation PR.

File: packages/ariakit-react-components/src/combobox/combobox-value.tsx (hunk 4 above)
Comment (diegohaz) [reply packages/ariakit-react-components/src/combobox/combobox-value.tsx:16]: Fixed in [`00cfb23`](https://github.com/ariakit/ariakit/commit/00cfb233767cb5d363a0f9a9c633800c172ae2e8): restored the original `ComboboxValue` description, `@see` link, and example while retaining the added `@deprecated` tag. Both reference pipelines now assert that the legacy description is preserved.

File: packages/ariakit-react-components/src/combobox/combobox.tsx (hunk 5 above)
Comment (diegohaz) [reply packages/ariakit-react-components/src/combobox/combobox.tsx]: Fixed in [`57ab279`](https://github.com/ariakit/ariakit/commit/57ab27923b6125cbbc300c77bcc48ab01ce4592c): renamed the memoized internal value to `inputValue` and kept the DOM boundary explicit as `value: inputValue`.

**Comment by diegohaz** [comment]
**Fixed**

- [`00cfb23`](https://github.com/ariakit/ariakit/commit/00cfb233767cb5d363a0f9a9c633800c172ae2e8) ([reply](https://github.com/ariakit/ariakit/pull/6900#discussion_r3663696830)): Removed the deprecation-only changeset while retaining the feature changeset.
- [`00cfb23`](https://github.com/ariakit/ariakit/commit/00cfb233767cb5d363a0f9a9c633800c172ae2e8) ([reply](https://github.com/ariakit/ariakit/pull/6900#discussion_r3663703645)): Restored the original `ComboboxValue` docs and added description assertions to both reference pipelines.
- [`57ab279`](https://github.com/ariakit/ariakit/commit/57ab27923b6125cbbc300c77bcc48ab01ce4592c) ([reply](https://github.com/ariakit/ariakit/pull/6900#discussion_r3663705597)): Renamed the memoized internal input value for consistency.

**Tracked**

- [#5695](https://github.com/ariakit/ariakit/issues/5695) ([reply](https://github.com/ariakit/ariakit/pull/6900#discussion_r3663698927), [reply](https://github.com/ariakit/ariakit/pull/6900#discussion_r3663700637)): Coordinating functional updater requests across both aliases belongs to the planned request-versus-commit store redesign; direct forwarding in the current synchronous model double-applies updates and a reentrant-safe paired write requires queued request semantics.
